src/common/resolveEnvVariables.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+
+type EnvObject = {
+  [variableName: string]: string;
+};
+
+//this function replaces all env variables within the docker compose yaml string with their values in the env file
+const resolveEnvVariables = (yamlText: string, filePath: string) => {
+  const envFileArray = filePath.split('/');
+  const envFilePath =
+    envFileArray.slice(0, envFileArray.length - 1).join('/') + '/.env';
+  //read envfile if there is one there
+  let envString: string;
+  try {
+    envString = fs.readFileSync(envFilePath).toString();
+  } catch (err) {
+    return yamlText;
+  }
+  let yamlTextCopy = yamlText;
+  console.log(envString);
+  //split by line
+  const envArray = envString.split('\n');
```

Comment:
This is not correct parsing of the .env file. Use https://github.com/motdotla/dotenv for more correct parsing, it's not as trivial as you might think. More specifically, this https://github.com/motdotla/dotenv#can-i-customizewrite-plugins-for-dotenv part might be more helpful